pkg/services/ngalert/notifier/channels/email_test.go

```diff
@@ -85,12 +85,15 @@ func TestEmailNotifier(t *testing.T) {
 				"Title":   "[FIRING:1]  (AlwaysFiring warning)",
 				"Message": "[FIRING:1]  (AlwaysFiring warning)",
 				"Status":  "firing",
-				"Alerts": template.Alerts{
-					template.Alert{
-						Status:      "firing",
-						Labels:      template.KV{"alertname": "AlwaysFiring", "severity": "warning"},
-						Annotations: template.KV{"runbook_url": "http://fix.me"},
-						Fingerprint: "15a37193dce72bab",
+				"Alerts": ExtendedAlerts{
+					ExtendedAlert{
+						Status:       "firing",
+						Labels:       template.KV{"alertname": "AlwaysFiring", "severity": "warning"},
+						Annotations:  template.KV{"runbook_url": "http://fix.me"},
+						Fingerprint:  "15a37193dce72bab",
+						SilenceURL:   "http:/localhost/alerting/silence/new?alertmanager=grafana&matchers=%2C%2Calertname%3DAlwaysFiring%2Cseverity%3Dwarning",
+						DashboardURL: "http:/localhost/d/abc",
+						PanelURL:     "http:/localhost/d/abc?viewPanel=5",
```

Comment:
Since the alerts are no longer linked to the dashboard, do we need this? Alternatively, should we not have a rule URL instead that points to the rule that created this alert using the rule ID in the labels?

Reply:
Rule no longer **must** be linked to panels, but they **can** be. This button was in the design :) 

There is currently no other "rule" view we can point to :-(